### ISSUE TITLE
Encountered issue when loading via shiny-server

### ISSUE THREAD

**Issue by happyshows** [edzer/sfr#249]
Hi,

I was able to install sf package fine in R console and be able to library it. However, when I try to library(sf) in Rmd file via shiny-server (open-source version). The file cannot be rendered correctly.

```
processing file: Demographic.Rmd
Error in dyn.load(file, DLLpath = DLLpath, ...) : 
  unable to load shared object '/usr/lib64/R/library/sf/libs/sf.so':
  libgdal.so.20: cannot open shared object file: No such file or directory
Quitting from lines 25-35 (Demographic.Rmd) 
Warning: Error in : package or namespace load failed for 'sf'
```

Since the file is already included in the path, I'm not sure what went wrong there.

```
echo $PATH
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/home/shows/proj/bin:/home/shows/gdal/bin:/root/bin

find / -name libgdal.so.20
/home/shows/gdal/lib/libgdal.so.20
```
Do I need to configure and argument when installing?
Appreciate the help...

**Comment by happyshows** [comment]
in my /home/shows/gdal/lib folder. I'm seeing the following

```
total 284488
drwxr-xr-x. 2 root root         6 Mar  3 18:11 gdalplugins
-rw-r--r--. 1 root root 206597218 Mar  3 21:23 libgdal.a
-rwxr-xr-x. 1 root root      1013 Mar  3 21:23 libgdal.la
lrwxrwxrwx. 1 root root        17 Mar  3 21:23 libgdal.so -> libgdal.so.20.1.1
lrwxrwxrwx. 1 root root        17 Mar  3 21:23 libgdal.so.20 -> libgdal.so.20.1.1
-rwxr-xr-x. 1 root root  84709705 Mar  3 21:23 libgdal.so.20.1.1
drwxr-xr-x. 2 root root        20 Mar  3 21:23 pkgconfig
```

I don't know why there's a symlink point to the libgdal.so.20.1.1. Is this the root cause?

**Comment by edzer** [comment]
Which OS do you work on? Having gdal in your PATH is not enough; on linux you'd need to set (export) `LD_LIBRARY_PATH=/home/shows/gdal/lib/`, or better, install gdal in an appropriate place like `/us/local/lib`, assuming you have permission to do so.

**Comment by happyshows** [comment]
I'm on RHEL 7. I already have LD_LIBRARY_PATH=/home/shows/gdal/lib/ configured.

Will creating a symlink under /usr/local/lib pointing to /home/shows/gdal work?

```
> Sys.getenv("LD_LIBRARY_PATH")
[1] "/usr/lib64/R/lib:/usr/lib/jvm/jre/lib/amd64/server:/usr/lib/jvm/jre/lib/amd64:/usr/lib/jvm/java/lib/amd64:/usr/java/packages/lib/amd64:/lib:/usr/lib::/home/shows/proj/lib:/home/shows/gdal/lib"
```

**Comment by edzer** [comment]
Maybe you have to run `ldconfig`? 

If you can create symlinks, why not install the libraries in `/usr/local/lib`, where they belong?

**Comment by happyshows** [comment]
Initially, I installed the library to /home/shows/gdal/lib

Below are my ldconfig info
```
 /sbin/ldconfig -p | grep gdal
	libgdal.so.1 (libc6,x86-64) => /lib64/libgdal.so.1
	libgdal.so (libc6,x86-64) => /lib64/libgdal.so
```

It seems that the library path is not being picked up in ldconfig. I did run it after the make install.

Shall I reinstall the entire gdal package? If so, what are the removal steps (for my existing installation)  so that I can have a clean slate. Thanks a lot!
